### PR TITLE
Add sample project on a low-resource language

### PR DESCRIPTION
**Pull request by ljvmiranda921**
## Description

This PR adds a project that trains a dependency parser for a low-resource language (tl). More notably, it demonstrates how to perform k-fold cross validation [as it is required for UD evaluation](https://universaldependencies.org/release_checklist.html#data-split).

Much of this is based [from this repository](https://github.com/ljvmiranda921/ud-tagalog-spacy) and [blog post](https://ljvmiranda921.github.io/notebook/2022/04/24/low-resource-dep-parse/). I didn't include the "training" and "packaging" steps since they're not relevant to what this project aims to demonstrate.   

## Checklist
- [x]  I confirm that I have the right to submit this contribution under the project's MIT license.
- [x]  I ran the tests, and all new and existing tests passed.
- [x]  My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
The project names use the component names, so I think would be better as `parser_low_resource`.

File: tutorials/dep_low_resource/scripts/kfold.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    '--n-folds'. It performs the split automatically, so you need to pass the
+    full corpus (not split into training/dev) in 'corpus_path'. Lastly,
+    we get the average of the scores for each fold to obtain the final metrics.
+    """
+    overrides = parse_config_overrides(ctx.args)
+    setup_gpu(use_gpu)
+
+    nlp = spacy.blank(lang)
+    doc_bin = DocBin().from_disk(corpus_path)
+    docs = list(doc_bin.get_docs(nlp.vocab))
```
Comment (adrianeboyd) [inline tutorials/dep_low_resource/scripts/kfold.py:68]: This isn't using the corpus reader from config so it doesn't include same options for augmentation, etc. that you might want. I'm trying to think about how to do this given that you don't have the typical setup with a train/dev split...

Comment (adrianeboyd) [reply tutorials/dep_low_resource/scripts/kfold.py:68]: Nevermind, I think was wrong about this.

File: tutorials/parser_low_resource/test_parser_low_resource.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from pathlib import Path
+
+from spacy.cli.project.assets import project_assets
+from spacy.cli.project.run import project_run
+
+
+def test_parser_low_resource_project():
+    root = Path(__file__).parent
+    project_assets(root)
+    project_run(root, "all", overrides={"vars.n_folds": 1, "vars.gpu": 0})
```
Comment (adrianeboyd) [inline tutorials/parser_low_resource/test_parser_low_resource.py:10]: The CI doesn't have a GPU:

```suggestion
    project_run(root, "all", overrides={"vars.n_folds": 1, "vars.gpu": -1})
```

(The tests are failing but this isn't being captured correctly by the new test script for some reason.)

Comment (adrianeboyd) [reply tutorials/parser_low_resource/test_parser_low_resource.py:10]: Actually, usually we use `-1` as the default GPU id for projects like this, so maybe that would be better here.

File: tutorials/README.md [hunk 3]
```diff
@@ -1,9 +1,10 @@
 <a href="https://explosion.ai"><img src="https://explosion.ai/assets/img/logo.svg" width="125" height="125" align="right" /></a>
 
-# 🪐 Project Templates: Tutorials (11)
+# 🪐 Project Templates: Tutorials (12)
 
 | Template | Description |
 | --- | --- |
+| [`dep_low_resource`](dep_low_resource) | Training a POS tagger and dependency parser for a low-resource language |
```
Comment (adrianeboyd) [inline tutorials/README.md:7]: Also update this README?

Comment (ljvmiranda921) [reply tutorials/README.md]: Updated in https://github.com/explosion/projects/pull/122/commits/646a868cd9fba78efc480972410eb16b8aaf7338

File: tutorials/parser_low_resource/test_parser_low_resource.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from pathlib import Path
+
+from spacy.cli.project.assets import project_assets
+from spacy.cli.project.run import project_run
+
+
+def test_parser_low_resource_project():
+    root = Path(__file__).parent
+    project_assets(root)
+    project_run(root, "all", overrides={"vars.n_folds": 2, "vars.gpu_id": -1}, capture=True)
```
Comment (adrianeboyd) [inline tutorials/parser_low_resource/test_parser_low_resource.py:10]:
```suggestion
    project_run(root, "all", overrides={"vars.n_folds": 2}, capture=True)
```

File: tutorials/parser_low_resource/project.yml [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  lang: "tl"
+  trg_treebank: "UD_Tagalog-TRG"
+  trg_data: "tl_trg-ud-test"
+  ugn_treebank: "UD_Tagalog-Ugnayan"
+  ugn_data: "tl_ugnayan-ud-test"
+
+assets:
+  - dest: "assets/${vars.trg_data}.conllu"
+    description: "Treebank data for UD_Tagalog-TRG"
+    url: "https://github.com/UniversalDependencies/UD_Tagalog-TRG/blob/master/tl_trg-ud-test.conllu"
```
Comment (adrianeboyd) [inline tutorials/parser_low_resource/project.yml:34]: Switch these to raw URLs?

Comment (ljvmiranda921) [reply tutorials/parser_low_resource/project.yml]: Hi, not sure if this is what we meant by raw URLs: https://github.com/explosion/projects/pull/122/commits/1d66eea9d426d2feaf1fad6745f6147a68bd6de9

**Review by svlandeg — COMMENTED** [review]
Looking good!

File: tutorials/parser_low_resource/project.yml [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        configs/${vars.config}.cfg 
+        --output-path metrics/${vars.ugn_data}-kfold.json 
+        --n-folds ${vars.n_folds}
+        --lang ${vars.lang} 
+        --system.seed ${vars.seed}
+        --use-gpu ${vars.gpu_id}
+    deps:
+      - "corpus/${vars.trg_data}.spacy"
+      - "corpus/${vars.ugn_data}.spacy"
+      - "configs/${vars.config}.cfg"
```
[the remark this answers is not in the document]
Comment (ljvmiranda921) [reply tutorials/parser_low_resource/project.yml:93]: Great point! Will do that.

File: tutorials/parser_low_resource/project.yml [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    outputs:
+      - "metrics/${vars.trg_data}-kfold.json"
+      - "metrics/${vars.ugn_data}-kfold.json"
+
+  - name: "clean"
+    help: "Remove intermediate files"
+    script:
+      - "rm -rf training/*"
+      - "rm -rf corpus/*"
```
[the remark this answers is not in the document]
Comment (ljvmiranda921) [reply tutorials/parser_low_resource/project.yml:102]: Yes that makes a lot of sense. Will do that as well :)

**Pull request merged**